module-utils/utility/CMakeLists.txt

```diff
@@ -28,6 +30,8 @@ target_link_libraries(utility
 
     PRIVATE
         hash-library::hash-library
+        purefs-paths
+        service-appmgr
```

Comment:
where is used `appmgr` in `utils`?

Reply:
I had to add `service-appmgr` to access `service-db/agents/settings/SystemSettings.hpp` because adding `serice-db` didn't resolve all dependencies.

Reply:
maybe it would be better to pass a string with the saved version number from the settings rather than the settings themselves? Please consider this

Reply:
I redesigned the implementation. If the db returns the default version then the `getPreviousVersionFromFile` function is invoked to get the version from the file. So `service-db/agents/settings/SystemSettings.hpp` isn't longer needed.